### PR TITLE
Premium themes: Remove one-off premium theme purchases (onboarding & theme showcase)

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Related to https://github.com/Automattic/wp-calypso/issues/73265

## Proposed Changes

* Remove the theme being added in the checkout when the user selects the free option
* Renamed `Or start with a free site subheader` to `Or select another theme`
* Point the user to the Design Picker if they clicked on the free option

## Testing Instructions

* Use the live link from the comments
* Go to `/start/with-theme?ref=calypshowcase&theme=yuga&premium=true&theme_type=premium`
* On the plans page, make sure that the subheader text is now changed to `Or select another theme` (make sure it also changes on mobile ports)
* When you click on the free option, you are now pointed to the Design Picker (instead of the checkout page)
* Start the flow again and this time select a domain
* Make sure that you are pointed to the checkout page and only the domain is added in the cart (No `Premium theme: Yuga` item is added)
* Open an incognito page and check that we no longer display the pricing in the `Pick design` button.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by mmtr** [comment]
Nice! This is heading into the right direction 👏 

However, I noted we don't show anything to users indicating that Premium themes require now a Premium plan:

Screenshot 2023-05-10 at 17 34 37

Should we use a banner similar to the one displayed for Marketplace themes?
Screenshot 2023-05-10 at 17 34 30
 

I also noted that for certain mobile resolutions (e.g. iPhone 12 Pro), the final dot after "start with another theme" is in a different line:

Screenshot 2023-05-10 at 17 36 36

Do we have any way to prevent that? Maybe using [the `preventWidows` util](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/formatting/prevent-widows.js)?

File: client/my-sites/theme/main.jsx [hunk 1]
```diff
@@ -1267,7 +1237,7 @@ class ThemeSheet extends Component {
 					onClick={ null === onClick ? noop : onClick }
 					href={ plansUrl }
 					showIcon={ true }
-					forceDisplay={ forceDisplay }
```
Comment (mmtr) [inline client/my-sites/theme/main.jsx]: This is introducing a regression for logged-in users. For instance, the upsell banner shouldn't be visible on WooCommerce themes for sites on the Business plan.

File: client/signup/steps/plans-theme-preselected/index.tsx [hunk 2]
```diff
@@ -40,7 +45,31 @@ function getHidePlanPropsBasedOnSignupDependencies(
...
+		subHeaderText = translate( 'Choose a plan or {{link}}start with another theme{{/link}}.', {
+			components: { link: freePlanButton },
+		} );
+	}
+
+	return (
+		<PlansStep
+			{ ...props }
+			{ ...hidePlanProps }
+			fallbackSubHeaderText={ preventWidows( subHeaderText ) }
```
Comment (mmtr) [inline client/signup/steps/plans-theme-preselected/index.tsx]: I still see the dot in a separate line:
Screenshot 2023-05-11 at 11 03 00

**Review by mmtr — COMMENTED** [review]
Thanks for adding the Premium plan upsell banner!

I noted this PR also removes the pricing label for logged-in users, but the premium theme product is still added to the shopping cart.

I guess we should keep the pricing label in this scenario, or expand the scope of this PR to prevent the premium theme from being added to the cart.

**Comment by BogdanUngureanu** [comment]
> I noted this PR also removes the pricing label for logged-in users, but the premium theme product is still added to the shopping cart.

Argh, good point. I had the impression that we don't display the pricing information for logged-in users. I think we can expand the scope of this PR to also cover the logged-in users. For logged-in users, it should be just a link change (instead of redirecting to checkout with a premium theme product, we'll be pointing them to checkout with a Premium subscription instead) :)

**Comment by mmtr** [comment]
One more thing!

When picking a Free theme in the logged-out showcase, we still display the "start with another theme" link.

Screenshot 2023-05-12 at 11 23 21

I think we should still say "start with a free site".

**Comment by BogdanUngureanu** [comment]
> When picking a Free theme in the logged-out showcase, we still display the "start with another theme" link.

Good catch, thanks! :)

**Review by mmtr — APPROVED** [review]
Excellent job, Bogdan! It works seamlessly

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7989703

Thank you @BogdanUngureanu for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7989703).

**Pull request merged**